lib/Model/Importer/Binary/SegmentImportHelpers.h

```diff
@@ -54,47 +53,45 @@ inline void importSymbolsInto(model::Binary &Binary,
       revng_assert(Offset.has_value());
 
       // Discard any symbols that would overlap an already existing one.
-      if (checkForOverlap(*Struct, *Offset, SymbolSize)) {
+      if (checkForOverlap(Struct, *Offset, SymbolSize)) {
         It = std::prev(DataSymbols.erase(It));
         continue;
       }
 
-      model::DefinitionReference T;
+      model::StructField Field;
+      Field.Offset() = *Offset;
+      Field.OriginalName() = SymbolName.str();
       if (SymbolSize == 1 || SymbolSize == 2 || SymbolSize == 4
           || SymbolSize == 8) {
-        T = Binary.getPrimitiveType(model::PrimitiveKind::Generic, SymbolSize);
+        Field.Type() = model::PrimitiveType::makeGeneric(SymbolSize);
       } else {
         // If the symbol has a non-standard size, make it an empty struct with
         // the same size instead.
-        T = createEmptyStruct(Binary, SymbolSize);
+        Field.Type() = Binary.makeStructDefinition(SymbolSize).second;
       }
 
-      model::QualifiedType FieldType = { T, {} };
-      model::StructField Field{ *Offset, {}, SymbolName.str(), {}, FieldType };
-      const auto &[_, Success] = Struct->Fields().insert(Field);
+      const auto &[_, Success] = Struct.Fields().insert(Field);
```

Comment:
I think you introduced `addField`, why not use it here?

Reply:
I guess because this code was converted before its introduction.

Either way, thanks for noticing.

Reply:
Done.